inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/special/ScheduledExecutorContextPropagationSensor.java

```diff
@@ -0,0 +1,82 @@
+package rocks.inspectit.oce.core.instrumentation.special;
+
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.asm.AsmVisitorWrapper;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.dynamic.DynamicType;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.springframework.context.annotation.DependsOn;
+import org.springframework.stereotype.Component;
+import rocks.inspectit.oce.bootstrap.Instances;
+import rocks.inspectit.oce.core.instrumentation.config.model.InstrumentationConfiguration;
+import rocks.inspectit.oce.core.instrumentation.context.ContextManagerImpl;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static net.bytebuddy.matcher.ElementMatchers.*;
+
+/**
+ * Special sensor for passing the context via the {@link ScheduledExecutorService}.
+ * This sensor will pass the context when using the following methods:
+ * <p><ul>
+ * <li>{@link ScheduledExecutorService#schedule(Runnable, long, TimeUnit)}
+ * <li>{@link ScheduledExecutorService#schedule(Callable, long, TimeUnit)}
+ * <li>{@link ScheduledExecutorService#scheduleWithFixedDelay(Runnable, long, long, TimeUnit)}
+ * <li>{@link ScheduledExecutorService#scheduleAtFixedRate(Runnable, long, long, TimeUnit)}
+ * </ul><p>
+ */
+@Component
+@DependsOn(ContextManagerImpl.BEAN_NAME)
+public class ScheduledExecutorContextPropagationSensor implements SpecialSensor {

```

Comment:
I am also getting this error:

```
2019-01-25 12:07:56,496 DEBUG 12730  --- [inspectIT] [pectit-thread-0] r.i.o.c.i.AsyncClassTransformer          : Applying instrumentation of class com.intellij.rt.execution.application.AppMainV2$1
2019-01-25 12:07:56,502 ERROR 12736  --- [inspectIT] [pectit-thread-0] r.i.o.c.i.InstrumentationManager         : Error retransforming classes!
java.lang.UnsupportedOperationException: class redefinition failed: attempted to change the class modifiers
	at sun.instrument.InstrumentationImpl.retransformClasses0(Native Method)
	at sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:144)
	at rocks.inspectit.oce.core.instrumentation.InstrumentationManager.checkClassesForConfigurationUpdates(InstrumentationManager.java:154)
	at rocks.inspectit.oce.core.service.BatchJobExecutorService.lambda$startJob$0(BatchJobExecutorService.java:126)
	at rocks.inspectit.oce.core.service.BatchJobExecutorService$BatchJob.lambda$new$0(BatchJobExecutorService.java:83)
	at io.grpc.Context$1.run(Context.java:595)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:747)
```

this could be related to the `Thread.start()` impl, let's analyze pls.. Maybe the `DEBUG` log meesage could print the active sensors?

Reply:
seems like we are instrumenting ourselves?

Reply:
ahh no, we correctly instrument the Java class used by us.. Anyway, the problem is something else!